common/browser_process.go

```diff
@@ -243,6 +244,9 @@ func (p *devToolsURLParser) err() error {
 		return p.errs[0]
 	}
 	if err := p.sc.Err(); err != nil {
+		if errors.Is(err, fs.ErrClosed) {
+			return fmt.Errorf("browser process shutdown unexpectedly before a connection could be made to it: %w", err)
+		}
```

Comment:
Please make the inner `if` unintended for a better line-of-sight ([see](https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html)):

```go
	err := p.sc.Err()
	if errors.Is(err, fs.ErrClosed) {
		return fmt.Errorf("browser process shutdown unexpectedly before a connection could be made to it: %w", err)
	}
	if err != nil {
		return fmt.Errorf("%w", err)
	}
```

The error message can be shorter without losing the intent:

`"browser process shutdown unexpectedly before establishing a connection: %w"`

Lastly, you don't need to wrap the error with `%w` ([see](https://go.dev/blog/go1.13-errors)). It might deserve a `nolint` directive when you don't wrap as the linter is not that smart.

Reply:
Yep, good suggestion. Resolved in 4d95e2ee63224b948bdd628064b02908e3a1497c